packages/slice-machine/src/features/settings/git/GitRepositoryConnectDialog.tsx

```diff
@@ -0,0 +1,86 @@
+import {
+  Box,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogHeader,
+  FormInput,
+  Text,
+} from "@prismicio/editor-ui";
+import type { GitRepo, GitRepoSpecifier } from "@slicemachine/manager";
+import { Formik } from "formik";
+import type { FC, ReactNode } from "react";
+import { toast } from "react-toastify";
+
+import { gitProviderToConfig } from "@src/features/settings/git/GitProvider";
+import { useWriteAPIToken } from "@src/features/settings/git/useWriteAPIToken";
+
+type GitRepositoryConnectDialogProps = {
+  linkRepo: (repo: GitRepoSpecifier) => Promise<void>;
+  repo: GitRepo;
+  trigger: ReactNode;
+};
+
+export const GitRepositoryConnectDialog: FC<
+  GitRepositoryConnectDialogProps
+> = ({ linkRepo, repo, trigger }) => {
+  const providerConfig = gitProviderToConfig[repo.provider];
+  const { updateToken } = useWriteAPIToken({ git: repo });
+  return (
+    <Dialog size={{ width: 448, height: "auto" }} trigger={trigger}>
+      <DialogHeader title="Prismic write API token required" />
+      <DialogContent>
+        <Formik
+          initialValues={{ writeAPIToken: "" }}
+          validate={(values) => {
+            if (values.writeAPIToken.length === 0) {
+              return { writeAPIToken: "Cannot be empty" };
+            }
+          }}
+          onSubmit={async (values) => {
+            try {
+              await updateToken(values.writeAPIToken);
+              await linkRepo(repo);
+            } catch (error) {
+              const message = `Could not connect to ${repo.name}`;
+              console.error(message, error);
+              toast.error(message);
+            }
+          }}
+        >
+          {(formikProps) => (
+            <form onSubmit={formikProps.handleSubmit}>
+              <Box flexDirection="column" gap={4} padding={16}>
+                <FormInput
+                  disabled={formikProps.isSubmitting}
+                  error={typeof formikProps.errors.writeAPIToken === "string"}
+                  label="Paste your write API token"
+                  onValueChange={(value) => {
+                    void formikProps.setFieldValue("writeAPIToken", value);
+                  }}
+                  placeholder="Write API token"
+                  size="large"
+                  value={formikProps.values.writeAPIToken}
+                />
+                <Text color="grey11">
+                  Required to update your Prismic repository with{" "}
+                  {providerConfig.name}.
```

Comment:
Would it be possible to link to the repository's settings page from here? Something like:

> Create a Write API Token

The design include a similar link: https://www.figma.com/file/wb7ujCJ4j7EiP7aDDGRYXB/SliceMachine%E2%84%A2-UI?type=design&node-id=11050-1083225&mode=design&t=3KZmbSzQdcHarVX9-4

A direct link is especially helpful now that the dialog covers the sidebar link to the repository.